disperser/common/semver/semver.go

```diff
@@ -28,7 +28,10 @@ func ScanOperators(operators map[core.OperatorID]*core.IndexedOperatorInfo, oper
 	operatorChan := make(chan core.OperatorID, len(operators))
 	worker := func() {
 		for operatorId := range operatorChan {
-			operatorSocket := core.OperatorSocket(operators[operatorId].Socket)
+			operatorSocket, err := core.ParseOperatorSocket(operators[operatorId].Socket)
+			if err != nil {
+				logger.Warnf("Failed to parse operator socket: %v", err)
```

Comment:
Error handling seems not right: the "operatorSocket" is still being used below even if this is error

Reply:
I think this needs to be fixed, a return value is undefined when there is error

Reply:
The return values are not undefined. These are empty strings, which is the zero value for strings.
The code currently (before structify PR) mandates this behavior. Unless we clean up this everywhere else, which I believe is not the scope of the PR. 

I think your concern is valid if we were returning a pointer and the dereferencing them errors would lead to panic. 
We don't do that here. The only use of a hypothetically errored out `operatorSocket` is to call `GetV1RetrievalSocket()` or `GetV1DispersalSocket()` which returns the zero value for this.

Reply:
I don't think this looks good. If you still want to proceed upon error, you may define and use an empty or zero value constant instead of using returned value.

Reply:
Again, this is the existing behavior, as this is a refactor I wouldn't add new functionality here. The easy solution here is to not make ParseOperatorSocket return an err, but that would be worse than what we have now.

Reply:
Sure this is existing, but why not make it less crazy (by having a `EmptyOperatorSocket` instead of using a return value upon error)